File: protocol/src/test/go/zfoogo/ObjectB.go
package protocol

type ObjectB struct {
	Flag bool
    InnerCompatibleValue int
}

func (protocol ObjectB) ProtocolId() int16 {
	return 103
}

func (protocol ObjectB) write(buffer *ByteBuffer, packet any) {
	if packet == nil {
	    buffer.WriteInt(0)
		return
	}
	var message = packet.(*ObjectB)
	var beforeWriteIndex = buffer.WriteOffset()
    buffer.WriteInt(4)
    buffer.WriteBool(message.Flag)
    buffer.WriteInt(message.InnerCompatibleValue)
    buffer.AdjustPadding(4, beforeWriteIndex)
}

func (protocol ObjectB) read(buffer *ByteBuffer) any {
	var packet = new(ObjectB)
	var length = buffer.ReadInt()
	if length == 0 {
		return packet
	}
	var beforeReadIndex = buffer.ReadOffset()
	var result0 = buffer.ReadBool()
    packet.Flag = result0
    if buffer.CompatibleRead(beforeReadIndex, length) {
        var result1 = buffer.ReadInt()
        packet.InnerCompatibleValue = result1
    }
	if length > 0 {
        buffer.SetReadOffset(beforeReadIndex + length)
    }
	return packet
}

File: protocol/src/test/csharp/zfoocs/Packet/ObjectB.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class ObjectB
    {
        public bool flag;
        public int innerCompatibleValue;
    }

    public class ObjectBRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 103;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            ObjectB message = (ObjectB) packet;
            int beforeWriteIndex = buffer.WriteOffset();
            buffer.WriteInt(4);
            buffer.WriteBool(message.flag);
            buffer.WriteInt(message.innerCompatibleValue);
            buffer.AdjustPadding(4, beforeWriteIndex);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            ObjectB packet = new ObjectB();
            bool result0 = buffer.ReadBool();
            packet.flag = result0;
            if (buffer.CompatibleRead(beforeReadIndex, length)) {
                int result1 = buffer.ReadInt();
                packet.innerCompatibleValue = result1;
            }
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}